### PR TITLE
Refactor reports to just have the queryArgs. Build the state url in the UI.

### PR DESCRIPTION
**Pull request by AdityaHegde**
Currently the dashboard url for a report is filled in by the UI. This is hard to manage when creating a report manually using file artifacts. Instead we can generate the dashboard url using the query args in the UI and redirect to that.

This PR adds,
- [x] Add a path `/<org>/<proj>/reports/<report>/open`. Map the query object to the url state and redirect to the dashboard.
- [x] Update backend to use this new path instead of using the url passed in annotations.
- [x] Loading spinner in the `/open` route.
- [x] Error handling in the `/open` route.
- [x] Cleanup of old code.

### CODE REVIEW

**Comment by nishantmonu51** [comment]
is this backwards compatible ?

**Comment by AdityaHegde** [comment]
It would take the query args from the older reports, ignoring the state url.

File: proto/rill/admin/v1/api.proto
Comment (begelundmuller) [inline proto/rill/admin/v1/api.proto:1069]: Can we make this a `google.protobuf.Timestamp` and then move the string formatting into the url builder in `externalURLs.reportOpen`?

**Review by begelundmuller — CHANGES_REQUESTED** [review]

**Review by begelundmuller — COMMENTED** [review]
Backend changes look good to me now!

**Comment by ericpgreen2** [comment]
I tested by creating a new report, clicking on the "Open in browser" button in the email, and I hit this error:
image

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (ericpgreen2) [inline web-admin/src/features/scheduled-reports/query-mapper.ts:48]: I'm noticing we have a lot of places in the codebase where we use a derived store to combine the results from two queries. Rather than a type like this one that mixes query status (`ready`, `error`) with query results (`state`, `metricsView`), it could be good if we created a generic type to hold the query status, like...

```typescript
type BaseQuery<T> = {
  isReady: boolean;
  error?: string
  data: T
}
```

which we could use to wrap the query results type...

```typescript
type DashboardState = {
    metricsView: string
    state: string
}

type DashboardStateQuery = BaseQuery<DashboardState>;
```

Comment (ericpgreen2) [reply web-admin/src/features/scheduled-reports/query-mapper.ts:48]: FWIW TanStack Query v5 will give us a first-class way to combine multiple queries, and the above approach mirrors TanStack.

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (ericpgreen2) [inline web-admin/src/features/scheduled-reports/query-mapper.ts:176]: Mind clarifying that these TODOs should be completed when we offer scheduled reports for the given query?

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (ericpgreen2) [inline web-admin/src/features/scheduled-reports/query-mapper.ts:141]: Leftover console log

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (ericpgreen2) [inline web-admin/src/features/scheduled-reports/query-mapper.ts:155]: The comment is helpful, but maybe even the function name could be `convertSnakeToCamelCase()`

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (ericpgreen2) [inline web-admin/src/features/scheduled-reports/query-mapper.ts:139]: Is it purposeful that the function returns `void`? Might be more clear if it returned the dashboard state object

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (ericpgreen2) [inline web-admin/src/features/scheduled-reports/query-mapper.ts:1]: Name suggestion: `query-mapper.ts` -> `get-dashboard-state-for-report.ts`

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (ericpgreen2) [inline web-admin/src/features/scheduled-reports/query-mapper.ts:54]: Name suggestion: `getDashboardStateForReport(...)`

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (ericpgreen2) [inline web-admin/src/features/scheduled-reports/query-mapper.ts:134]: Name suggestion: `getDashboardState`

**Review by ericpgreen2 — COMMENTED** [review]
Overall, code is clear! Mostly left some name and type recommendations.

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (AdityaHegde) [reply web-admin/src/features/scheduled-reports/query-mapper.ts:176]: UI only adds MetricsViewComparison right now. Need to clarify if we need to add support for others in the 1st cut. Lemme double check with product before merging this PR.

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (AdityaHegde) [reply web-admin/src/features/scheduled-reports/query-mapper.ts:139]: It updated the dashboard in place. It felt like an overkill to make it copy again and return it, since we are creating the dashboard object before passing it to the mapper.

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (AdityaHegde) [reply web-admin/src/features/scheduled-reports/query-mapper.ts:48]: Good point. I will not refactor other places but update this derived state.

**Comment by ericpgreen2** [comment]
I just created a report and tried to open it, but now I hit a different error: 
image

The intended report view: 
image

File: web-common/src/features/compound-query-result.ts
Comment (ericpgreen2) [inline web-common/src/features/compound-query-result.ts:1]: I think it'd be better if this lived in `web-common/src/runtime-client` or `web-common/src/lib`

File: web-admin/src/features/scheduled-reports/query-mapper.ts
Comment (ericpgreen2) [reply web-admin/src/features/scheduled-reports/query-mapper.ts]: Probably not needed for the first cut because we don't yet mention Report YAML in our docs anywhere.

My point was that the TODOs confused me until I realized that we don't yet offer scheduled reports for these queries. A note like 'TODO: Scheduled reports not yet available for this query' could help clarify.

**Review by ericpgreen2 — APPROVED** [review]
Works for me now! Approving so you can merge after fixing CI tests.

**Review by begelundmuller — APPROVED** [review]

**Pull request merged**
